bundles/org.openhab.binding.somfymylink/src/main/java/org/openhab/binding/somfymylink/internal/handler/SomfyShadeHandler.java

```diff
@@ -0,0 +1,116 @@
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.BridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@NonNullByDefault
+public class SomfyShadeHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SomfyShadeHandler.class);
+
+    public SomfyShadeHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        initDeviceState();
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        logger.debug("Bridge status changed to {} updating {}", bridgeStatusInfo.getStatus(), getThing().getLabel());
+
+        if (bridgeStatusInfo.getStatus() == ThingStatus.ONLINE
+                && getThing().getStatusInfo().getStatusDetail() == ThingStatusDetail.BRIDGE_OFFLINE) {
+            initDeviceState();
+        } else if (bridgeStatusInfo.getStatus() == ThingStatus.OFFLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    }
+
+    public void initDeviceState() {
+        Bridge bridge = getBridge();
+
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No bridge configured");
+            logger.debug("Initialized device state for shade {} {}", ThingStatus.OFFLINE,
+                    ThingStatusDetail.CONFIGURATION_ERROR);
+        } else if (bridge.getStatus() == ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.ONLINE);
+            logger.debug("Initialized device state for shade {}", ThingStatus.ONLINE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            logger.debug("Initialized device state for shade {} {}", ThingStatus.OFFLINE,
+                    ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (CHANNEL_SHADELEVEL.equals(channelUID.getId())) {
+                String targetId = channelUID.getThingUID().getId();
+
+                if (command instanceof RefreshType) {
+                    return;
+                }
+
+                if (command instanceof UpDownType) {
+                    if (command.equals(UpDownType.DOWN)) {
+                        getBridgeHandler().commandShadeDown(targetId);
+                    } else {
+                        getBridgeHandler().commandShadeUp(targetId);
+                    }
+                }
+
+                if (command instanceof StopMoveType) {
+                    getBridgeHandler().commandShadeStop(targetId);
+                }
```

Comment:
Why aren't these command methods implemented in this class?

Reply:
i put them in the bridge handler so that all communication is handled via the bridge class. All the commands etc... and build and handle there.